packages/flutter/lib/src/services/flutter_version.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @docImport 'dart:io'
+
+/// [FlutterVersion] contains various meta information about with which
+/// Flutter version this app was compiled.
+/// It tries to mimic the information that is seen when executing `flutter --version`.
+///
+/// When this Flutter version was build from a fork, or when Flutter runs in a
+/// custom embedder, these values might be unreliable.
```

Comment:
What goes wrong with these values if built from a fork? I'd expect them to continue to work consistently with `flutter --version`.

Reply:
A fork could change the tool code and remove the code which adds these values as Dart defines.